homeassistant/components/lyric/__init__.py

```diff
@@ -189,6 +173,6 @@ def device_info(self) -> DeviceInfo:
         return {
             "connections": {(dr.CONNECTION_NETWORK_MAC, self._mac_id)},
             "manufacturer": "Honeywell",
-            "model": self._device_model,
-            "name": self._device_name,
+            "model": self.device.deviceModel,
```

Comment:
I don't see `self.device` defined.

Reply:
It's here:

https://github.com/home-assistant/core/blob/531c6bed8177da1f53456935bdc2d552d0741254/homeassistant/components/lyric/__init__.py#L161-L164

Reply:
The coordinator holds the location, which again holds the device, there is no need to pass location or device to the class when the coordinator is passed to it.